### PR TITLE
Issue #8828: G now also gives real solutions

### PR DESCRIPTION
**Pull request by mihir-w**
In`solve` function, `got_s` is a function which stores the symbols for which we have already solved. If another equation is encountered in terms of symbols solved, it discards that solution. `got_s` was declared _outside_ the result for loop, letting solutions drop. Just brought the definition inside. Now G returns the real solutions.

Regarding extra solutions with `manual=True`, it seems the problem is in `solve_poly_system` which implements Groebner basis approach ( I don't know much about it. ). Since it seemed like a different issue, I created pull request for this separate fix. 

Added test in test_solver.py by creating `test_issue_8828` function.

Also please forgive for the unnecessary commits , am new to git and got confused.  ( Edit: I think the commit is fine now. If otherwise, please inform. )

@smichr please review and suggest.

### CODE REVIEW

File: sympy/solvers/tests/test_solvers.py [hunk 1]
```diff
@@ -747,6 +747,34 @@ def test_issue_5767():
...
+    x1 = 0
+    y1 = -620
+    r1 = 920
+    x2 = 126
+    y2 = 276
+    x3 = 51
+    y3 = 205
+    r3 = 104
+
+    f1 = (x-x1)**2 + (y-y1)**2 - (r1-z)**2
```
Comment (smichr) [inline sympy/solvers/tests/test_solvers.py]: Nice! What a relief to see this fixed. :-)

Could you just apply a little PEP8 spacing here: we try to put space around `+ and -` and put a space after commas.

File: sympy/solvers/tests/test_solvers.py [hunk 2]
```diff
@@ -747,6 +747,34 @@ def test_issue_5767():
...
+    G = g1,g2,g3
+
+    A = solve(F,[x,y,z])
+    B = solve(G,[x,y,z])
+
+    p = [[i.evalf() for i in j] for j in A]
+    q = [[i.evalf() for i in j] for j in B]
+
+    for sol in p:
+        assert( sol in q )
```
Comment (smichr) [inline sympy/solvers/tests/test_solvers.py]: can you just `assert set(p) == set(q)`?

Comment (mihir-w) [inline sympy/solvers/tests/test_solvers.py]: `G` is returning 4 solutions. 2 imaginary and 2 real. At first I thought that the 2 imaginary solutions were not detected by `F`, but on testing it seems that the 2 imaginary solutions are wrong. Checking of solutions is not proper. Trying to work around, but other tests keep failing.

**Comment by mihir-w** [comment]
`G` was returning additional wrong solutions when one of the equation was not polynomial in nature ( Regardless of manual = True/False ). Hence added an extra check after the final solutions have been calculated.
Also added formatting for `test_issue_8828`.

So now, both `F` and `G` return the right solutions without returning any extra solutions.

File: sympy/solvers/tests/test_solvers.py (hunk 2 above)
Comment (smichr) [inline sympy/solvers/tests/test_solvers.py]: I'll send a PR

**Comment by smichr** [comment]
[Please see this PR](https://github.com/mihir-w/sympy/pull/1).

**Comment by smichr** [comment]
Nice work!

**Comment by mihir-w** [comment]
Regarding the [PR](https://github.com/mihir-w/sympy/pull/1), I merged and ran the tests on my computer. Few tests failed, either they were not simplified, or because of the removal of `checksol` testing.

I think that intermittent testing through `checksol` is necessary. So should I re-add the checksol testing? 
Also regarding the simplification, should I simplify the results or change the test files accordingly ?

**Comment by smichr** [comment]
I sent a new PR...it's marked as wip b/c I don't have time to look at it now. Feel free to modify. I wonder if the got_s was in the right place formerly, so moved it back. I could be wrong.

**Comment by mihir-w** [comment]
oh! I messed up. Any idea how to clean up?
sorry!

**Pull request closed without merging**